tests/unit/test_polynomialembeddor.py

```diff
@@ -0,0 +1,343 @@
+# Copyright 2016 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+
+import unittest
+
+from collections import defaultdict
+
+from dwave.embedding.polynomialembeddor import processor, _bulk_to_linear, _to_linear, _bulk_to_chimera
+from dwave.embedding.polynomialembeddor import _chimera_neighbors, random_processor, _to_chimera
+
+
+def verify_chains(Proc, emb):
+    proc = Proc._proc0
+    for i, e in enumerate(emb):
+        for u in e:
+            assert u in proc, "qubit %s doesn't exist" % (u,)
+        for u, v in zip(e, e[1:]):
+            assert (u in proc[v]) and (
+                v in proc[u]), "chain is not a connected subgraph"
+        E = set(e)
+        for f in emb[:i]:
+            assert not any(v in E for v in f), "qubit appears in two chains!"
+
+
+def verify_clique(Proc, emb, size, length):
+    verify_chains(Proc, emb)
+    proc = Proc._proc0
+
+    assert len(emb) == size, "expected a K_{%s}" % size
+    for i, e in enumerate(emb):
+        assert len(e) == length, "expected chain lengths to be %s but found a chain of length %s" % (
+            length, len(e))
+        for f in emb[:i]:
+            assert any(
+                u in proc[v] for u in e for v in f), "no coupler between %s and %s" % (e, f)
+
+
+def verify_biclique(Proc, emb, num_A, num_B, len_A, len_B):
+    proc = Proc._proc0
+
+    A, B = emb
+    verify_chains(Proc, A)
+    verify_chains(Proc, B)
+
+    assert (len(A) == num_A) and (len(
+        B) == num_B), "expected a K_{%s,%s} but got a K_{%s,%s}" % (num_A, num_B, len(A), len(B))
+    assert all(len(e) == len_A for e in A) and all(
+        len(e) == len_B for e in B), "expected chain lengths %s and %s" % (len_A, len_B)
+
+    for e in A:
+        for f in B:
+            assert any(
+                u in proc[v] for u in e for v in f), "no coupler between %s and %s" % (e, f)
+
+
+class TestEden(unittest.TestCase):
+    def setUp(self):
+
+        M = 12
+        N = 7
+        L = 4
+
+        eden_qubits = [(x, y, u, k) for x in range(M)
+                       for y in range(N) for u in (0, 1) for k in range(L)]
+
+        # one K_12, contains K_{8,8}
+        Cliq1 = {(0, 0), (1, 0), (2, 0), (0, 1), (1, 1), (0, 2)}
```

Comment:
I think the "Cliq*" and "Rect*" are variables. Therefore, I would have expected them to be lowercase.

Reply:
This is just one out of _many, many, many_ PEP-8 inconsistencies in the `polynomialembeddor.py` (and related code). But since that submodule is already semi-private, and it would be a lot of work to "fix it" just for the sake of consistency, it's probably best to just leave it as is. Also, see [here](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds). :slightly_smiling_face:

Reply:
Yes, that was my thought. Applies to most/all the comments on that file.

Reply:
That code is lifted from https://github.com/dwavesystems/chimera-embedding

Reply:
true true... this project predates my participation in pep8 by a couple of years ;)